### DISCUSSION TITLE
error in accessing the detection annotations

### DISCUSSION THREAD

**Discussion by DrMRizk** [rafaelpadilla/review_object_detection_metrics#76] [Q&A]
Hello @rafaelpadilla 
Thank you for the tool.
I am using YOLO anotation format [class-id, x_center, y_center, w, h] relative values
example: 
----------------------------------------------------------
3 0.510547 0.647917 0.811719 0.362500
7 0.505859 0.327778 0.952344 0.352778
----------------------------------------------------------

For the prediction I followed the first choice:
[class-id, confidence, x_center, y_center, w, h] 
example:
----------------------------------------------------------
3 99.6900 0.8953 0.1283 0.0415 0.0670
3 99.8200 0.5152 0.4906 0.6199 0.7697
----------------------------------------------------------

The tool can access the ground truth and provide correct statistics.
However, for the detections, only few lables are accessed. Only the annotations of 11 images are counted. The statistics is not correct. Accordingly,  the output metric result is not true.
The following message appears in the shell: 
"Warning: Image not found in the directory None. It is required to get its dimensions"

I added the ground truths images to the folder annotations of the predictions, even though it is not recommended. The same error still occus and the same warning message appears in the shell.

Note that, annotations (text files) in the GT and predictions have the same names.

I attached the annotations of the predeictions 
[predicted_labels.zip](https://github.com/rafaelpadilla/review_object_detection_metrics/files/7268065/predicted_labels.zip)
and the annotations of the GT
[gt_lables.zip](https://github.com/rafaelpadilla/review_object_detection_metrics/files/7268395/gt_lables.zip)

Thank you in advance

**Comment by adavid3-unb** [comment]
I am facing the same issue with YOLOv5 annotations. 
In my case, none of the detections are being recognized by the tool.

@DrMRizk have you been able to solve this issue?

**Reply by adavid3-unb** [reply]
Hello @DrMRizk @rafaelpadilla,
I'm not sure if you were able to resolve this issue already, but I was able to identify the cause of the problem and resolve it on my end. I also checked your attached sample data to make sure it is the case.

The metrics tool has an issue. The issue arises from prediction files that have more than one dot (".") in the filename. The tool misidentifies everything that is coming after the dot as the file extension and omits it. 
For example, one of your prediction files is called "CAPT.JIMMY_T.MORAN_2016-02-05_08-58.txt", and has multiple dots in addition to the extension.

I believe this is happening in the **find_file()** function that is found in the **general_utils.py** class. Specifically, the **splitext()** function seems to be mishandling those filenames.

A temporary workaround is to rename predictions filenames that contain dots to something else. Make sure to also rename the corresponding ground truth images and labels.

I hope this helps!
